test/e2e-upgrade-tests.sh

```diff
@@ -58,41 +58,55 @@ function knative_setup() {
   install_latest_release
 }
 
-# Script entry point.
+function test_upgrade() {
+  header "Running preupgrade tests with $CLIENT_TYPE client"
 
-initialize $@ --skip-istio-addon
+  go_test_e2e -tags=preupgrade -timeout=${TIMEOUT} ./test/upgrade/${CLIENT_TYPE} \
+    --resolvabledomain=$(use_resolvable_domain) || fail_test
 
-# TODO(#2656): Reduce the timeout after we get this test to consistently passing.
-TIMEOUT=10m
+  header "Starting prober test"
 
-header "Running preupgrade tests"
+  # Remove this in case we failed to clean it up in an earlier test.
+  rm -f /tmp/prober-signal
 
-go_test_e2e -tags=preupgrade -timeout=${TIMEOUT} ./test/upgrade \
-  --resolvabledomain=$(use_resolvable_domain) || fail_test
+  go_test_e2e -tags=probe -timeout=${TIMEOUT} ./test/upgrade \
+    --resolvabledomain=$(use_resolvable_domain) &
+  PROBER_PID=$!
+  echo "Prober PID is ${PROBER_PID}"
 
-header "Starting prober test"
+  install_head
 
-# Remove this in case we failed to clean it up in an earlier test.
-rm -f /tmp/prober-signal
+  header "Running postupgrade tests"
+  go_test_e2e -tags=postupgrade -timeout=${TIMEOUT} ./test/upgrade/${CLIENT_TYPE} \
+    --resolvabledomain=$(use_resolvable_domain) || fail_test
 
-go_test_e2e -tags=probe -timeout=${TIMEOUT} ./test/upgrade \
-  --resolvabledomain=$(use_resolvable_domain) &
-PROBER_PID=$!
-echo "Prober PID is ${PROBER_PID}"
+  # The prober is blocking on /tmp/prober-signal to know when it should exit.
+  #
+  # This is kind of gross. First attempt was to just send a signal to the go test,
+  # but "go test" intercepts the signal and always exits with a non-zero code.
+  echo "done" > /tmp/prober-signal
 
-install_head
+  header "Waiting for prober test"
+  wait ${PROBER_PID} || fail_test "Prober failed"
+}
+# Script entry point.
 
-header "Running postupgrade tests"
-go_test_e2e -tags=postupgrade -timeout=${TIMEOUT} ./test/upgrade \
-  --resolvabledomain=$(use_resolvable_domain) || fail_test
+initialize $@ --skip-istio-addon
 
-# The prober is blocking on /tmp/prober-signal to know when it should exit.
-#
-# This is kind of gross. First attempt was to just send a signal to the go test,
-# but "go test" intercepts the signal and always exits with a non-zero code.
-echo "done" > /tmp/prober-signal
+# TODO(#2656): Reduce the timeout after we get this test to consistently passing.
+TIMEOUT=10m
+
+
+# Running upgrade test using v1alpha1 client
+CLIENT_TYPE=v1alpha1
+test_upgrade
```

Comment:
I suggest passing client type as a parameter, it's easier to read and has no side effects.

Reply:
According to the new pull request the script doesn't have any changes.